### PR TITLE
tag indexing bug

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

**Comment by ymatsuda** [comment]
@yingjieMiao @leogrim 
Please review the change. I found a place that we fail to notify collection update.

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala [hunk 1]
```diff
@@ -53,56 +53,51 @@ class UriIntegrityActor @Inject()(
...
   /** tricky point: make sure (user, uri) pair is unique.  */
   private def handleBookmarks(oldUserBookmarks: Map[Id[User], Seq[Bookmark]], newUriId: Id[NormalizedURI])(implicit session: RWSession) = {
+
     val deactivatedBms = oldUserBookmarks.map{ case (userId, bms) =>
       val oldBm = bms.head
       bookmarkRepo.getByUriAndUserAllStates(newUriId, userId) match {
         case None => {
           log.info(s"going to redirect bookmark's uri: (userId, newUriId) = (${userId.id}, ${newUriId.id}), db or cache returns None")
           bookmarkRepo.deleteCache(oldBm)     // NOTE: we touch two different cache keys here and the following line
-          bookmarkRepo.save(oldBm.withNormUriId(newUriId)); None
```
Comment (ymatsuda) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala:62]: This, returning None, is where we miss collection update.

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala:62]: wow, great catch!

**Comment by yingjieMiao** [comment]
lgtm!

**Comment by yingjieMiao** [comment]
trick is collection-uri relation is changed without a change in collection-bookmark relation.

File: kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/integrity/UriIntegrityPlugin.scala:62]: :+1: sharp observation!

**Pull request merged**